### ISSUE TITLE
Custom VNET change breaks normal template generation

### ISSUE THREAD

**Issue by karataliu** [Azure/acs-engine#1569]
PR #1150 adds 'VIRTUAL_NETWORK_RESOURCE_GROUP' in parameter. 

When not using custom vnet, the `VIRTUAL_NETWORK_RESOURCE_GROUP` has an empty value, which causes every follow up parameters to shift one position ahead.

Here is the part of the result based on commit `4df8a77c9`:
/etc/kubeletes/azure.json
```
 "cloud":"none", # value for network policy
 "vnetResourceGroup": "k8s-master-20487671-routetable", # value for route table
 "routeTableName": "agentpool1-availabilitySet-20487671", # value for availability set
```

/etc/default/kubelet
```
KUBELET_MAX_PODS=false # value for CLOUDPROVIDER_BACKOFF
```

Here is the process run snapshot from another run:
```
root      2712  0.0  0.0   4528   776 ?        S    03:00   0:00 /bin/sh -c /usr/bin/nohup /bin/bash -c "/bin/bash /opt/azure/containers/provision.sh <guid> <guid> <rg> <location> k8s-subnet k8s-master-16231024-nsg k8s-vnet-16231024  k8s-master-16231024-routetable agentpool1-availabilitySet-16231024
```

Note that there are two spaces between 'k8s-vnet-16231024' and 'k8s-master-16231024-routetable', that's the empty value of 'VIRTUAL_NETWORK_RESOURCE_GROUP'.

**Comment by karataliu** [comment]
cc @sylr @jackfrancis  
This happens when '/opt/azure/containers/provision.sh' runs.


Suppose we should find new ways for passing parameter to the provisioning script, currently updating or debugging such a numeric identified parameter is painful and error-prone.

**Comment by karataliu** [comment]
Fixed by #1575
